code/game/objects/items/grenades/ghettobomb.dm

```diff
@@ -159,3 +139,92 @@
 	update_mob()
 	explosion(src.loc,-1,-1,2, flame_range = 4)	// small explosion, plus a very large fireball.
 	qdel(src)
+
+
+// Molotov Cocktail
+/obj/item/reagent_containers/food/drinks/bottle/molotov
+	name = "molotov cocktail"
+	desc = "A empty bottle with a rag in it. Needs to be filled with a flammable liquid before being lit on fire and chucked at someone you don't like."
+	icon = 'icons/fallout/objects/guns/explosives.dmi'
+	lefthand_file = 'icons/fallout/onmob/weapons/special_lefthand.dmi'
+	righthand_file = 'icons/fallout/onmob/weapons/special_righthand.dmi'
+	icon_state = "molotov"
+	item_state = "ied"
+	list_reagents = list()
+	var/list/accelerants = list(
+		/datum/reagent/consumable/ethanol,
+		/datum/reagent/fuel,
+		/datum/reagent/clf3,
+		/datum/reagent/phlogiston,
+		/datum/reagent/napalm,
+		/datum/reagent/hellwater,
+		/datum/reagent/toxin/plasma,
+		/datum/reagent/toxin/spore_burning,
+		)
+	var/active = FALSE
+
+/obj/item/reagent_containers/food/drinks/bottle/molotov/CheckParts(list/parts_list)
+	. = ..()
+	var/obj/item/reagent_containers/food/drinks/bottle/bottle_found = locate() in contents
+	if(!bottle_found)
+		return
+	icon_state = bottle_found.icon_state
+	bottle_found.reagents.copy_to(src, 100)
+	if(!bottle_found.isGlass)
+		desc = "[desc] You're not sure if making this out of a carton was the brightest idea."
+		isGlass = FALSE
+
+/obj/item/reagent_containers/food/drinks/bottle/molotov/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	var/firestarter = FALSE
+	for(var/datum/reagent/reagent_in_bottle in reagents.reagent_list)
+		for(var/accelerant_type in accelerants)
+			if(istype(reagent_in_bottle, accelerant_type))
```

Comment:
Can you make a type cache?
Look for examples in the code of it.
This is a relevant proc: `is_type_in_typecache()`

Reply:
I could figure it out given time, currently I don't quite grasp the implementation here, if its not needed we might just forego it, and leave the rest as is?